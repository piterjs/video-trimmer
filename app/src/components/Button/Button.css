.Button {
    background-color: #fff;
    border-color: #dbdbdb;
    border-width: 1px;
    color: #363636;
    cursor: pointer;
    justify-content: center;
    padding-bottom: calc(.375em - 1px);
    padding-left: .75em;
    padding-right: .75em;
    padding-top: calc(.375em - 1px);
    text-align: center;
    white-space: nowrap;
    user-select: none;
    transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}
.Button:focus {
    outline: none;
}
.Button:disabled {
    box-shadow: none;
    opacity: .5;
    cursor: not-allowed;
}

