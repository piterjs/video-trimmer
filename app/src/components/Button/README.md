# Button

- **Tag**: May be used as Button or Link 
- **Type**: May be button or submit type
- **Color**: May be default (white), primary (green) or secondary (blue)
- **Size**: May be small, normal, medium or large size
- **Disabled**: May be disabled

---

**TODO**
- add text button
- add outlined button
- add icon-button
- add button-with-icon (left, right)
- add pressed state for button
- add rounded button
