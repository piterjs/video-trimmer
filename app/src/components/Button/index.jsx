import React from 'react';

import './Button.css';
import './colors/default.css';
import './colors/primary.css';
import './colors/secondary.css';

import '../Control/Control.css';
import '../Control/sizes.css';

// TODO: add text button
// TODO: add outlined button
// TODO: add icon-button
// TODO: add button-with-icon (left, right)
// TODO: add pressed state for button
// TODO: add rounded button

function Button ({size = 'normal', color = 'default', tag = 'button', className = '', ...rest}) {
    const classes = [
        'Control',
        `Control--size-${size}`,
        'Button',
        `Button--color-${color}`,
        className
    ].join(' ').trim()

    const Tag = `${tag}`;

    return <Tag
        className={classes}
        {...rest}
    >{rest.children}</Tag>
}

export default Button;
