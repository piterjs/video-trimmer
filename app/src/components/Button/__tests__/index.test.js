import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../';

describe('Button', () => {
    it('should be defined', () => {
        expect(Button).toBeDefined();
    });
    it('should render correctly', () => {
        const tree = renderer
        .create(<Button />)
        .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
