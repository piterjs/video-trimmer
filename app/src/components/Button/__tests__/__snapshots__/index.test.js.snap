// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  className="Control Control--size-normal Button Button--color-default"
/>
`;
